AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -25,6 +26,41 @@ static void runLoopSourceCallback(void *info) {
 #endif
 }
 
+#pragma mark - ASDeallocThread
+
+@interface ASDeallocThread : NSThread
+@property (nonatomic, strong) ASRunLoopQueue *deallocQueue;
+@end
+
+@implementation ASDeallocThread
+
+- (void)main
+{
+  self.name = @"ASDeallocThread";
+  [self deallocQueue];
+  while (YES) {
+    // This method will still return once all sources have finished.
+    [[NSRunLoop currentRunLoop] runUntilDate:[NSDate distantFuture]];
```

Comment:
@maicki @Adlai-Holler looking for tips here.  I can't get this code right and it's driving me nuts.  Even though this line is called repeatedly, it does NOT result in the running of the observer's handler block.

Am I missing something obvious?  Is there a better way to keep the thread alive and runloop idling while still allowing the thread to sleep most of the time?

I'm considering ditching the runloop and just having the timer to periodically check the queue.  In this case, a separate class from ASRunloopQueue (but sharing the same file due to related purpose) would be simplest.  Especially if we can get a lockless queue in place.